### ISSUE TITLE
Problems with Filter\Search

### ISSUE THREAD

**Issue by southeasterntech** [Ylianst/MeshCentral#6065]
Problem:
Search is not working for many users

We have been having problems with multiple users with search\filter not working at all.  For one user (happens to be admin) it's working fine.
-We have tried in incognito mode : FAIL
-Tried granting user admin access : FAIL
-Tried different browsers : FAIL
-Tried reverting to 1.1.0, tried upgrading to 1.1.22 : FAIL
-Promote regular user to Admin and it does not resolve so, seems not related to permissions: FAIL

Nothing seems to fix it

**Expected behavior**
Filter should work for all users

**Screenshots**
![image](https://github.com/Ylianst/MeshCentral/assets/45638572/be37a9be-a5c4-4ad8-bbfd-78ba04d78c57)
![image](https://github.com/Ylianst/MeshCentral/assets/45638572/e4cd905d-cecc-4285-a196-1384de72e661)
![image](https://github.com/Ylianst/MeshCentral/assets/45638572/a9200669-646c-4778-a62a-ad4a5d6e0956)

**Server Software (please complete the following information):**
 - OS: Amazon EC-2 \Linux
 - Virtualization: ?
 - Network: WAN
 - Version: 1.1.22
 - Node: 16.14.0

**Client Device (please complete the following information):**
 - Device: Desktops\Windows
 - OS:Win10
 - Network: LAN device connecting to Mesh server at AWS
 - Browser:Multiple Browsers
 - MeshCentralRouter Version: [if applicable]

**Additional context**
Latest Log error:

-------- 5/1/2024, 3:13:52 PM ---- 1.1.0 --------

node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module './lib/types/urlencoded'
Require stack:
- /home/ec2-user/node_modules/meshcentral/node_modules/body-parser/index.js
- /home/ec2-user/node_modules/meshcentral/meshcentral.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at loadParser (/home/ec2-user/node_modules/meshcentral/node_modules/body-parser/index.js:150:16)
    at Function.get [as urlencoded] (/home/ec2-user/node_modules/meshcentral/node_modules/body-parser/index.js:122:12)
    at serverStart (/home/ec2-user/node_modules/meshcentral/webserver.js:6355:50)
    at /home/ec2-user/node_modules/meshcentral/webserver.js:307:17
    at /home/ec2-user/node_modules/meshcentral/db.js:2259:115
    at /home/ec2-user/node_modules/mongodb/lib/utils.js:371:9
    at /home/ec2-user/node_modules/mongodb/lib/cursor/abstract_cursor.js:260:32
    at /home/ec2-user/node_modules/mongodb/lib/cursor/abstract_cursor.js:510:55
    at /home/ec2-user/node_modules/mongodb/lib/utils.js:371:9
    at /home/ec2-user/node_modules/mongodb/lib/sessions.js:136:24
    at maybePromise (/home/ec2-user/node_modules/mongodb/lib/utils.js:357:5)
    at ClientSession.endSession (/home/ec2-user/node_modules/mongodb/lib/sessions.js:133:41) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/ec2-user/node_modules/meshcentral/node_modules/body-parser/index.js',
    '/home/ec2-user/node_modules/meshcentral/meshcentral.js'
  ]

**Your config.json file**
```
{
	"settings": {
		"MongoDb": "mongodb://127.0.0.1:27017/meshcentral",
		"selfupdate": true,
		"Cert": "mesh.south*********",
		"Port": 443,
		"WANonly": true,
		"IgnoreAgentHashCheck": false,
		"RedirPort": 80,
                "RelayPort": 44453,
		"browserPong": 30,
		"agentPong": 30,
		"agentPing": 30,
		"dbexpire": {
			"events": 1728000,
			"powerevents": 864000,
			"statsevents": 2592000
		},
		"DesktopMultiplex": true
	},
	"letsencrypt": {
		"email": "info@south******",
		"names": "mesh.south******",
		"rsaKeySize": 3072,
		"production": true
	},
	"domains": {
		"": {
			"agentCustomization": {
				"image": "setechnical-mesh.png",
				"installText": "Welcome to the Southeastern Technical Agent Installer. Please click install or update to continue.",
				"certurl": "https://mesh.south*****"
			},
			"altmessenging": [{
					"name": "Knowledge Base",
					"url": "http://docs.south******/"
				},
				{
					"name": "Inventory",
					"url": "https://inv.setech*****/"
				},
{
					"name": "Compliance",
					"url": "https://compliance.setech****/"
				},
				{
					"name": "Jitsi Video Call",
					"url": "https://meet.jit.si/setech***"
				}
			],
			"title": "",
			"titlePicture": "title-southeastern.png",
			"newAccounts": 0,
			"UserSessionIdleTimeout": 60,
			"title2": "",
			"Minify": true,
			"SiteStyle": 2,
			"footer": "Level Up Your IT",
                        "LoginKey": "SeTech",
			"sessionRecording": {
				"onlySelectedDeviceGroups": true,
				"index": true,
				"_maxRecordings": 500,
				"maxRecordingSizeMegabytes": 250,
				"protocols": [1, 2, 200]
			}
		}
	},
	"smtp": {
		"host": "smtp-relay.sendinblue.com",
		"port": 465,
		"from": "mesh@south*****",
		"user": "mesh@south****",
		"pass": "*********",
		"tls": true
	}
}

**Comment by si458** [comment]
That's weird?
So the search doesn't work with normal users without admin permissions
But the user with admin permission and it works?

P.s can u create a separate issue for the blank my devices page as that's a separate issue, and edit this issue to remove that issue

**Comment by southeasterntech** [comment]
Yes I’ll create a separate issue. 
 
 But I promoted one of the normal users to admin and it still didn’t work!
 
 
      Thanks,
       
  Shane D. Lewis
         
         On May 1, 2024, at 12:41 PM, Simon Smith ***@***.***> wrote:
  
  
    ﻿ That's weird?
   So the search doesn't work with normal users without admin permissions
   But the user with admin permission and it works? P.s can u create a separate issue for the blank my devices page as that's a separate issue, and edit this issue to remove that issue —

**Comment by southeasterntech** [comment]
Also, just saw, that for these users offline\online filter also does not work

**Comment by si458** [comment]
You can only search using a few different search things
https://ylianst.github.io/MeshCentral/meshcentral/devicetabs/#search-or-filter

**Comment by southeasterntech** [comment]
In this case, we're just searching for groups, so g:acme would bring up the acme device group.
-Additionally noted that for these users, the List\Thumbnail buttons do not work.

**Comment by southeasterntech** [comment]
This issue was simultaneously solved with the solution to #6066.  Thanks Team!
